### ISSUE TITLE
Barrier doesn't work if the workspace which Barrier is on is not in use

### ISSUE THREAD

**Issue by panos** [debauchee/barrier#689]
### Operating Systems ###

Server: Debian Sid

Client: Gentoo Linux (amd64 17.1)

### Barrier Version ###

2.3.2

### Steps to reproduce bug ###

1. Set up a working connection between client and server
2. [Change workspace](](https://help.gnome.org/users/gnome-help/stable/shell-workspaces-switch.html.en)

### Other info ###

* When did the problem start to occur? **See "Steps to reproduce bug"**
* Is there a way to work around it? **No**
* Does this bug prevent you from using Barrier entirely? **No**

Just in case I didn't make it clear enough, the server's desktop environment in question is Gnome.

**Comment by p12tic** [comment]
Let's not close valid bug reports.

**Comment by Rockburner** [comment]
I'm getting the same issue.
Server: Ubuntu 22 (using Wayland)
Client Ubuntu 20 (using Unity)

Symptom:  Mouse can only travel to Client from 1 workspace on server.
Mouse can travel BACK to server regardless of which workspace server is currently showing
Interesting quirk: if Mouse is on Client desktop, switching workspaces on Server seems to disable 'SHIFT' key for client, until mouse has gone back to Server and then back to client.
Also - using keyboard to switch workspaces only works on server, even when mouse is on client desktop.

**Comment by symposion** [comment]
Yes, I also have this issue. Arch Gnome Wayland on both ends. Waynergy as a client.
